pkg/cloudprovider/providers/openstack/openstack_instances.go

```diff
@@ -81,6 +88,41 @@ func (i *Instances) List(name_filter string) ([]types.NodeName, error) {
 	return ret, nil
 }
 
+// GetAllZones returns all the zones from current servers in same region
+func (os *OpenStack) GetAllZones() (sets.String, error) {
+	// TODO: Caching, but this is currently only called when we are creating a volume,
+	// which is a relatively infrequent operation, and this is only # zones API calls
+	zones := sets.NewString()
+
+	// retrieve all servers
+	opts := servers.ListOpts{}
+	// Retrieve a pager (i.e. a paginated collection)
+	compute, err := openstack.NewComputeV2(os.provider, gophercloud.EndpointOpts{
+		Region: os.region,
+	})
+	if err != nil {
+		glog.Warningf("Failed to find compute endpoint: %v", err)
+		return nil, err
+	}
+	pager := servers.List(compute, opts)
```

Comment:
Yep this is what I am thinking as well. In my opinion the best option could be that we could fetch az information from node labels? So it means that we should somehow query v1.Nodes in that GetAllZones(). But otherwise imo that AWS solution is not much better, because it is using server tags to see is this master or not? In case of openstack that information is not added to anywhere? 

If we modify this part of code I think we should somehow use that v1.Nodes, and use node labels to fetch information. That is also TODO in AWS part, so they thought that same as well.